google-cloud-firestore/src/main/java/com/google/cloud/firestore/RecursiveDelete.java

```diff
@@ -160,48 +187,59 @@ public void onNext(DocumentSnapshot snapshot) {
           }
 
           public void onError(Throwable throwable) {
-            lastError =
-                new FirestoreException("Failed to fetch children documents.", Status.UNAVAILABLE);
-            onQueryEnd();
+            synchronized (lock) {
+              String message = "Failed to fetch children documents. " + throwable.getMessage();
+              lastError = new FirestoreException(message, Status.UNAVAILABLE);
```

Comment:
Java exception "bubbling" usually involves passing the original exception as the cause to the wrapping exception.